### PR TITLE
Provider on job preview [FLYW-694]

### PR DESCRIPTION
**Pull request by ehlertjd**
Determines what the effective provider for a job should be, and returns that provider's record.
When using for analysis gears, set the destination as the parent container of the analysis.

### Review Checklist

- Tests were added to cover all code changes
- Documentation was added / updated
- Code and tests follow standards in CONTRIBUTING.md
- Extra care is taken and log messages are created for [Critical Data](https://github.com/flywheel-io/core/wiki/Critical-Data-Flows)

### CODE REVIEW

File: api/jobs/handlers.py [hunk 1]
```diff
@@ -560,6 +561,23 @@ def add(self):
 
         return { '_id': job.id_ }
 
+    @require_login
+    def determine_provider(self):
+        """Determine the effective provider for a job"""
+        payload = self.request.json
```
Comment (hkethi002) [inline api/jobs/handlers.py]: Should we be validating the job input here against the schema?

Comment (ehlertjd) [reply api/jobs/handlers.py]: Less worried about it here than in the [add endpoint](https://github.com/flywheel-io/core/blob/78d3ab00976c0e8482586c8ec9aef61caf86325e/api/jobs/handlers.py#L547-L562). At least here we don't put anything into the database.

Comment (hkethi002) [reply api/jobs/handlers.py]: Ah, I forgot that enqueueing a job doesn't actually enqueue it

**Comment by ehlertjd** [comment]
Good point, lets address that in a separate refactor PR

**Pull request merged**